lms/static/js/learner_dashboard/views/course_enroll_view.js

```diff
@@ -20,23 +20,105 @@
+                    if(this.urlModel){
+                        this.trackSelectionUrl = this.urlModel.get('track_selection_url');
+                        this.dashboardUrl = this.urlModel.get('dashboard_url');
+                        this.verificationUrl = this.urlModel.get('id_verification_url');
                     }
                 },
 
+                updateIsEnrolled: function(){
+                    
+                },
+
                 render: function() {
-                    var filledTemplate = this.tpl(this.model.toJSON());
-                    HtmlUtils.setHtml(this.$el, filledTemplate);
+                    var filledTemplate;
+                    if (this.$parentEl &&
+                        this.enrollModel &&
+                        this.model.get('course_key')){
+
+                        filledTemplate = this.tpl(this.model.toJSON());
+                        HtmlUtils.setHtml(this.$el, filledTemplate);
+                        HtmlUtils.setHtml(this.$parentEl, HtmlUtils.HTML(this.$el));
+                        this.$('.run-select').val(this.model.get('run_key'));
+                    }
                 },
 
                 handleEnroll: function(){
                     //Enrollment click event handled here
+                    if(!this.model.get('is_enrolled')){
+                        // actually enroll
+                        this.enrollModel.save({
+                            course_id: this.model.get('course_key')
+                        }, {
+                            success: _.bind(this.enrollSuccess, this),
+                            error: _.bind(this.enrollError, this)
+                        });
+                    }
+                },
+
+                handleRunSelect: function(event){
+                    var runKey;
+                    if(event.target){
+                        runKey = $(event.target).val();
+                        if(runKey){
+                            this.model.updateRun(runKey);
+                        }
+                    }
+                },
+
+                enrollSuccess: function(){
+                    var track = this.model.get('mode_slug'),
+                        courseKey = this.model.get('course_key');
+                    if (track) {
+                        if ( track !== 'honor' &&
+                            track !== 'audit' &&
+                            this.verificationUrl ) {
+                            
+                            // Go to the start of the verification flow
+                            this.redirect( this.verificationUrl + courseKey + '/' );
+                        }
+                    } else if (this.trackSelectionUrl) {
+                        // Go to track selection page
+                        this.redirect( this.trackSelectionUrl + courseKey );
+                    }
+
+                    this.model.set({
+                        is_enrolled: true
+                    });
```

Comment:
So if we are redirecting there is no point in updating the model (because it's currently not synced with the backend)? If so I suggest wrapping this in the else statement from the previous if...if else for clarity.